### PR TITLE
feat!: Better button styles and development environment for styling

### PR DESCRIPTION
**Pull request by cobraz**
Add `gap` to buttons (in order to support icons, etc) and adds hot-reload to `yarn dev` using NextJS.

BREAKING CHANGE: Removes a few button variants (`close`, `icon`, `empty`, `large`).

### CODE REVIEW

File: src/theme.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export const theme = {
       alignItems: 'center',
       justifyContent: 'center',
       cursor: 'pointer',
+      gap: 2,
```
Comment (cobraz) [inline src/theme.ts:49]: Using `gap` with a flexbox is pretty cutting-edge. See: https://caniuse.com/?search=gap

**Comment by braaar** [comment]
> lso add `gap` to primary button (in order to support icons, etc)

So, in other words, only the primary button supports the icons properly?

File: pages/index.tsx [hunk 2]
```diff
@@ -82,11 +83,21 @@ const DocsPage = () => (
...
+                {key} btn
+              </Button>
+            </Box>
+          ))}
+          <Box>
+            <Button sx={{ mr: 3, mb: 3 }}>
+              Hello
+              <HiArrowRight />
+            </Button>
+          </Box>
```
Comment (braaar) [inline pages/index.tsx:99]:
```suggestion
          {Object.keys(theme.buttons).map(key => (
            <Flex sx={{flexDirection: 'column', gap: 3, mb: 3}} key={key}>
              <Button variant={key} sx={{ mr: 3}}>
                {key} btn
              </Button>
            <Button variant={key} sx={{ mr: 3}}>
              {key} btn
              <HiArrowRight />
            </Button>
          </Flex>
          ))}
          
```

We should test using icons for every variant, in my opinion.

File: src/theme.ts (hunk 1 above)
Comment (braaar) [reply src/theme.ts:49]: I would say so.

**Comment by cobraz** [comment]
> > lso add `gap` to primary button (in order to support icons, etc)
> 
> So, in other words, only the primary button supports the icons properly?

No, all of them supports it now (added in https://github.com/indivorg/theme/pull/15/commits/8d3968c448df62c7a819277f2c520aeed416fd59)

**Review by braaar — APPROVED** [review]

**Pull request merged**